### PR TITLE
UCP/KEEPALIVE: fixed diagnostic of ep_check return value

### PR DESCRIPTION
**Pull request by hoopoepg**
- added new allowed return value for ep_check - UCS_ERR_ENDPOINT_TIMEOUT,
  it is returned when EP is in failed state

### CODE REVIEW

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -2315,7 +2315,9 @@ void ucp_ep_do_keepalive(ucp_ep_h ep, ucp_lane_map_t *lane_map)
         status = uct_ep_check(ep->uct_eps[lane], 0, NULL);
         if (status == UCS_OK) {
             *lane_map &= ~UCS_BIT(lane);
-        } else if (status != UCS_ERR_NO_RESOURCE) {
+        } else if ((status != UCS_ERR_NO_RESOURCE) &&
+                   (status != UCS_ERR_ENDPOINT_TIMEOUT)) {
```
Comment (brminich) [inline src/ucp/core/ucp_ep.c:2319]: maybe we need to return UCS_OK from UCT instead, so this ep would be immediately removed from the lane map?

Comment (hoopoepg) [reply src/ucp/core/ucp_ep.c]: ok, returned success

**Review by brminich — APPROVED** [review]
btw, ep_check should not be invoked for failed ep. When can it happen?

**Comment by hoopoepg** [comment]
uct_mm_ep_check set EP to failed status immediately

**Review by yosefe — APPROVED** [review]

**Pull request merged**